### PR TITLE
update _dispatchEventFirefoxIE for closure

### PR DESCRIPTION
**Pull request by garlicnation**

### CODE REVIEW

**Comment by notwaldorf** [comment]
Looks like https://github.com/PolymerElements/iron-input/pull/86 was already merged hat did the same thing?

**Comment by justinfagnani** [comment]
I think this is a better fix than suppressing the error

**Comment by justinfagnani** [comment]
@garlicnation can you revert the change from #86 in this PR?

File: iron-input.html [hunk 1]
```diff
@@ -137,16 +136,22 @@
...
+    _dispatchEventFirefoxIE: function(event) {
       // Due to Firefox bug, events fired on disabled form controls can throw
       // errors; furthermore, neither IE nor Firefox will actually dispatch
       // events from disabled form controls; as such, we toggle disable around
       // the dispatch to allow notifying properties to notify
       // See issue #47 for details
       var disabled = this.disabled;
       this.disabled = false;
-      this._origDispatchEvent.apply(this, arguments);
+      var defaultPrevented = this._origDispatchEvent.call(this, event);
```
Comment (danbeam) [inline iron-input.html]: `apply` is arguably (heh) just as safe and more flexible

**Comment by danbeam** [comment]
LGTM this is what #86 wants to be when its grows up

File: iron-input.html (hunk 1 above)
Comment (justinfagnani) [inline iron-input.html]: isn't this equivalent to `var defaultPrevented = this._origDispatchEvent(event);`?

Comment (danbeam) [inline iron-input.html]: yes

**Comment by garlicnation** [comment]
@justinfagnani done

File: iron-input.html (hunk 1 above)
Comment (justinfagnani) [inline iron-input.html]: @garlicnation I'd change this to a simple invoke before merging

**Comment by justinfagnani** [comment]
LGTM++

**Comment by e111077** [comment]
LGTM

File: iron-input.html [hunk 2]
```diff
@@ -137,16 +136,22 @@
...
+    _dispatchEventFirefoxIE: function(event) {
       // Due to Firefox bug, events fired on disabled form controls can throw
       // errors; furthermore, neither IE nor Firefox will actually dispatch
       // events from disabled form controls; as such, we toggle disable around
       // the dispatch to allow notifying properties to notify
       // See issue #47 for details
       var disabled = this.disabled;
       this.disabled = false;
-      this._origDispatchEvent.apply(this, arguments);
+      var defaultPrevented = this._origDispatchEvent(event);
```
Comment (danbeam) [inline iron-input.html]: out of curiosity, what happens if this `dispatchEvent` call has the side effect changing `this.disabled`?

**Pull request merged**